src/runtime.rs

```diff
@@ -0,0 +1,72 @@
+// This file is Copyright its original authors, visible in version control history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. You may not use this file except in
+// accordance with one or both of these licenses.
+
+use tokio::task::JoinHandle;
+
+use std::future::Future;
+
+pub(crate) struct Runtime {
+	mode: RuntimeMode,
+}
+
+impl Runtime {
+	pub fn new() -> Result<Self, std::io::Error> {
+		let mode = match tokio::runtime::Handle::try_current() {
+			Ok(handle) => RuntimeMode::Handle(handle),
```

Comment:
It still makes me feel a bit uncomfortable that that handle is capture here invisibly, and that there's the implicit assumption that the user will keep this alive.

Removing it here, and letting the user pass it in themselves via the builder seems to be a more transparent way to signal that it is used and needs to remain available.